### PR TITLE
Add "link" icon to headings with ✨ JavaScript ✨ 

### PR DESCRIPTION
**Pull request by mcmillan**
See alphagov/tech-docs-template#11 for background.

This adds some JavaScript to inject an anchor into each heading tag on
page load. The rationale here is that ultimately if this doesn't work
due to the user having JS disabled or any other reason, it won't break
the core experience of the website – it's an enhancement.

The alternative, rendering these in Ruby, introduces a reasonable amount
of technical debt and also raises some questions around accessibility.

### CODE REVIEW

**Comment by mcmillan** [comment]
@36degrees This has a set of specs in now, so probably good to review whenever you fancy.

File: spec/javascripts/anchored-heading-spec.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  var $element;
+  var $heading;
+
+  beforeEach(function() {
+    module = new GOVUK.Modules.AnchoredHeadings();
+    $element = $('<div>\
+      <h1 id="caas" data-module="anchored-headings">Cake as a service</h1>\
+    </div>');
+    $heading = $element.find('h1');
+    module.start($element);
```
Comment (36degrees) [inline spec/javascripts/anchored-heading-spec.js]: Should the data-module attribute not be on the div, as in the real world(™)?

File: source/javascripts/govuk/modules.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+(function($, root) {
```
Comment (36degrees) [inline source/javascripts/govuk/modules.js]: Assuming this file has come from _somewhere else_, so not looking to closely at it.

File: source/javascripts/modules/anchored-headings.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  Modules.AnchoredHeadings = function() {
+    this.start = function($element) {
+      var headings = $element.find('h1, h2, h3, h4, h5, h6');
+      headings.each(injectAnchor);
+    };
+
+    function injectAnchor() {
+      var $this = $(this);
+      var fragment = '#' + $this.attr('id');
```
Comment (36degrees) [inline source/javascripts/modules/anchored-headings.js]: Can save a variable and a concatenation with:

```js
$this.prepend(
    '<a href="#' + $this.attr('id') + '" class="anchored-heading__icon" aria-hidden="true">🔗</a>'
);
```

Unless you think the extra step aids legibility?

File: spec/javascripts/anchored-heading-spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  'use strict';
+
+  var module;
+  var $element;
+  var $heading;
+
+  beforeEach(function() {
+    module = new GOVUK.Modules.AnchoredHeadings();
+    $element = $('<div>\
+      <h1 id="caas" data-module="anchored-headings">Cake as a service</h1>\
```
Comment (36degrees) [inline spec/javascripts/anchored-heading-spec.js]: 🍰

**Comment by mcmillan** [comment]
@36degrees Good notes, duly update 👌

**Comment by 36degrees** [comment]
We possibly should've firmed up the link icon before this got merged. Not a big issue, just need to make sure it gets changed before it starts being used anywhere!

**Pull request merged**